torch/utils/_mode_utils.py

```diff
@@ -0,0 +1,69 @@
+from typing import Iterator
+from dataclasses import dataclass
+
+
+# in order to dedupe the logic between python mode and torch_function mode, this
+# is a container to hold all the differences between the modes. Then functions like
+# _enable_mode are able to use this container to call functions or get correctly
+# formatted names
+@dataclass
+class _ModeInfo:
+    mode_name: str
+    mode_class: type  # the class related to the mode that's allowed to be passed in
+    base_mode_class: type  # the base class of mode_class that dispatches to the original function
+    required_fn: str  # string version of the function required, either torch_function or torch_dispatch
```

Comment:
Actually, I'm confused why these are fields. These fields (or at least `mode_class`) don't appear to be part of the public API of `_ModeInfo`, and they're only used in the downstream abstract method implementations. If they don't need to be in the public API, it seems clearer and more straightforward for the downstream subclass to take care of organizing things as necessary. As it stands, these fields seem quite unmotivated.